Taarafo.Portal.Web.Tests.Unit/Services/PostViews/PostViewServiceTests.Logic.RetrieveAll.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using Taarafo.Portal.Web.Models.Posts;
+using Taarafo.Portal.Web.Models.PostViews;
+using Xunit;
+
+namespace Taarafo.Portal.Web.Tests.Unit.Services.PostViews
+{
+    public partial class PostViewServiceTests
+    {
+        [Fact]
+        public async Task ShouldRetrieveAllPostViewsAsync()
+        {
+            // given
+            DateTimeOffset randomDateTime = GetRandomDateTime();
+            var randomAuthorId = Guid.NewGuid();
+
+            dynamic dynamicPostProperties =
+                CreateRandomPostProperties(
+                    auditDates: randomDateTime,
+                    auditIds: randomAuthorId);
+
+            var post = new Post
+            {
+                Id = dynamicPostProperties.Id,
+                Content = dynamicPostProperties.Content,
+                CreatedDate = dynamicPostProperties.CreatedDate,
+                UpdatedDate = dynamicPostProperties.UpdatedDate,
+                Author = dynamicPostProperties.Author
+            };
+
+            var postView = new PostView
+            {
+                Id = dynamicPostProperties.Id,
+                Content = dynamicPostProperties.Content,
+                CreatedDate = dynamicPostProperties.CreatedDate,
+                UpdatedDate = dynamicPostProperties.UpdatedDate,
+                Author = dynamicPostProperties.Author
+            };
+
+            var randomPosts = new List<Post> { post };
```

Comment:
We need to have an actual list of posts. Here's a good example:
https://github.com/hassanhabib/OtripleS.Portal/pull/44

Reply:
@hassanhabib Thank you for posting the right reference. I had have looked on an older one. Great improvement checking random count, instead of only one entry in collection.